Sitefinity CLI/Commands/UpgradeCommand.cs

```diff
@@ -51,447 +45,97 @@ internal class UpgradeCommand
-            {
-                this.SolutionPath = Path.GetFullPath(this.SolutionPath);
-            }
-
-            if (!File.Exists(this.SolutionPath))
-            {
-                throw new FileNotFoundException(string.Format(Constants.FileNotFoundMessage, this.SolutionPath));
-            }
-
-            var additionalPackagesIds = this.AdditionalPackagesString?.Split(",", StringSplitOptions.RemoveEmptyEntries).Select(x => x.Trim()).ToList();
-
-            if (additionalPackagesIds != null && additionalPackagesIds.Any() && additionalPackagesIds.Any(x => !this.allowedAdditionalPackagesIds.Contains(x)))
-            {
-                throw new ArgumentException($"The given additional packages cannot be upgraded. The currently supported additional packages for upgrade are: {string.Join(", ", this.allowedAdditionalPackagesIds)}");
-            }
+            await Validate();
 
-            if (!this.SkipPrompts && !this.promptService.PromptYesNo(Constants.UpgradeWarning))
-            {
-                this.logger.LogInformation(Constants.UpgradeWasCanceled);
-                return;
-            }
+            this.logger.LogInformation(Constants.SearchingProjectForReferencesMessage);
 
-            if (string.IsNullOrEmpty(this.Version))
-                this.SetLatestVersion();
-
-            this.logger.LogInformation("Searching the provided project/s for Sitefinity references...");
-
-            var sitefinityProjectFilePaths = this.GetProjectsPathsFromSolution(this.SolutionPath, true);
-            var projectsWithouthSitefinityPaths = this.GetProjectsPathsFromSolution(this.SolutionPath).Except(sitefinityProjectFilePaths);
+            IEnumerable<string> sitefinityProjectFilePaths = this.projectService.GetProjectPathsFromSolution(this.SolutionPath, this.Version, true);
 
             if (!sitefinityProjectFilePaths.Any())
             {
                 Utils.WriteLine(Constants.NoProjectsFoundToUpgradeWarningMessage, ConsoleColor.Yellow);
                 return;
             }
 
-            Dictionary<string, string> configsWithoutSitefinity = this.GetConfigsForProjectsWithoutSitefinity(projectsWithouthSitefinityPaths);
-
             this.logger.LogInformation(string.Format(Constants.NumberOfProjectsWithSitefinityReferencesFoundSuccessMessage, sitefinityProjectFilePaths.Count()));
-            this.logger.LogInformation(string.Format("Collecting Sitefinity NuGet package tree for version \"{0}\"...", this.Version));
-            var packageSources = await this.packageSourceBuilder.GetNugetPackageSources(this.NugetConfigPath);
+            this.logger.LogInformation(string.Format(Constants.CollectionSitefinityPackageTreeMessage, this.Version));
 
-            NuGetPackage newSitefinityPackage = await this.sitefinityPackageManager.GetSitefinityPackageTree(this.Version, packageSources);
+            NuGetPackage installedPackage = await this.packageService.InstallPackage(UpgradeOptions, sitefinityProjectFilePaths);
```

Comment:
For better readability we can construct the parameter we pass to the Install package method here. Also it is strange the naming is odd. This method is doing more that just installing the package.